src/v/iceberg/rest_client/types.h

```diff
@@ -17,9 +17,14 @@
 #include <seastar/core/sstring.hh>
 
 #include <ada/expected.h>
+#include <boost/beast/http/status.hpp>
 
 namespace iceberg::rest_client {
 
+// An http call related error, either a status code or a string describing an
+// exception caught during the call.
+using http_call_error = std::variant<boost::beast::http::status, ss::sstring>;
```

Comment:
Would it be better to surface the actual `std::exception_ptr` as the second variant instead of `ss::sstring`. that way you can always check the type of the exception if needed, and you don't only have a string which is really only useful for logging.